pkg/config/config_test.go

```diff
@@ -0,0 +1,439 @@
+	}
+}
+
+func TestConfigParseHCLFilesValidLokocfgFilePathExpand(t *testing.T) {
+	hclconfig := `
+variable "ssh_public_keys" {}
+
+backend "local" {
+ path = "test/backend/path"
+}
+cluster "packet" {
+  asset_dir = pathexpand("~")
+  cluster_name = "test-cluster"
+  version = "latest"
+  channel = "stable"
+  network_mtu = 1480
+  pod_cidr = "10.2.0.0/16"
+  service_cidr = "10.3.0.0/16"
+  count = 1
+  ssh_pubkeys = var.ssh_public_keys
+  tags = {
+    "key1"= "test1"
+    "key2" = "test2"
+  }
+}
+`
+	varconfig := `
+ssh_public_keys="test-ssh-key"
+`
+
+	hclconfigMap := map[string][]byte{
+		"test.lokocfg": []byte(hclconfig),
+	}
+	varconfigMap := map[string][]byte{
+		"test.vars": []byte(varconfig),
+	}
+
+	config, diags := ParseHCLFiles(hclconfigMap, varconfigMap)
+	if diags.HasErrors() {
+		for _, diag := range diags {
+			t.Error(diag)
+		}
+
+		t.Fatal("loading valid HCL configuration should not produce any errors:")
+	}
+
+	if config.ClusterConfig == nil {
+		t.Fatalf("config should not be nil after loading valid HCL configuration")
+	}
+}
+
+func TestConfigParseHCLFilesValidLokocfgFileWithFilepath(t *testing.T) {
+	hclconfig := `
+backend "local" {
+ path = "test/backend/path"
+}
+cluster "packet" {
+  asset_dir = "test/path"
+  cluster_name = "test-cluster"
+  version = "latest"
+  channel = "stable"
+  network_mtu = 1480
+  pod_cidr = "10.2.0.0/16"
+  service_cidr = "10.3.0.0/16"
+  count = 1
+  ssh_pubkeys = file("./config_test.go")
+  tags = {
+    "key1"= "test1"
+    "key2" = "test2"
+  }
+}
+`
+	varconfig := `
+ssh_public_keys="test-ssh-key"
+`
+
+	hclconfigMap := map[string][]byte{
+		"test.lokocfg": []byte(hclconfig),
+	}
+	varconfigMap := map[string][]byte{
+		"test.vars": []byte(varconfig),
+	}
+
+	config, diags := ParseHCLFiles(hclconfigMap, varconfigMap)
+	if diags.HasErrors() {
+		t.Fatalf("loading valid HCL configuration should not produce any errors: %q", diags)
+	}
+
+	if config.ClusterConfig == nil {
+		t.Fatalf("config should not be nil after loading valid HCL configuration")
+	}
+}
+func TestConfigParseHCLFilesMultipleLokocfgFiles(t *testing.T) {
+	hclconfig := `
+variable "ssh_public_keys" {}
+
+cluster "packet" {
+  asset_dir = "test/path"
+  cluster_name = "test-cluster"
+  version = "latest"
+  channel = "stable"
+  network_mtu = 1480
+  pod_cidr = "10.2.0.0/16"
+  service_cidr = "10.3.0.0/16"
+  count = 1
+  ssh_pubkeys = var.ssh_public_keys
+  tags = {
+    "key1"= "test1"
+    "key2" = "test2"
+  }
+}
+`
+	hclconfig2 := `
+backend "local" {
+ path = "test/backend/path"
+}
+`
+	varconfig := `
+ssh_public_keys="test-ssh-key"
+`
+	hclconfigMap := map[string][]byte{
+		"cluster.lokocfg": []byte(hclconfig),
+		"backend.lokocfg": []byte(hclconfig2),
+	}
+	varconfigMap := map[string][]byte{
+		"cluster.vars": []byte(varconfig),
+	}
+
+	config, diags := ParseHCLFiles(hclconfigMap, varconfigMap)
+	if diags.HasErrors() {
+		t.Fatalf("config should load from multiple config files: %s", diags)
```

Comment:
If we don't test for the parsed content, how this is testing that multiple files has been parsed?

Reply:
True, we can check if the names are set properly or not 
i.e `local` and `packet`.